Training24/TestCoinReturn.c

```diff
@@ -0,0 +1,73 @@
+// ------------------------------------------------------------------------------------
+// Training ~ A training program for new joiners at Metamation, Batch - July 2024.
+// Copyright (c) Metamation India.
+// Dhaarani Mohanasundaram
+// -------------------------------------
+// TestCoinReturn.c
+// Program on Test2.2 branch.
+// ------------------------------------------------------------------------------------
+#include <stdio.h>
+#include <stdlib.h>
+#include <time.h>
+#include "CoinReturn.h"
+
+#define RED "\x1b[31m"
+#define GREEN "\x1b[32m"
+#define RESET "\x1b[0m"
+
+void TestCoinReturn (int testNumber, int paidAmount, int actualAmount, int expectedCoins[], int* coins, int numCoins) {
+   int coinsReturned[5] = { 0 };
+   CalculateChange (paidAmount, actualAmount, coinsReturned, coins, numCoins);
+   if (coinsReturned[0] == -1) {
+      int isCoinsMatch = (expectedCoins[0] == -1);
+      printf ("Test %d: %s\nPaid amount: %d, Actual amount: %d; Insufficient amount paid.\n\n",
+         testNumber, isCoinsMatch ? GREEN "Passed" RESET : RED "Failed" RESET, paidAmount, actualAmount);
+      return;
+   }
+   int isCoinsMatch = 1;
+   for (int i = 0; i < numCoins + 1; i++) {
+      if (coinsReturned[i] != expectedCoins[i]) {
+         isCoinsMatch = 0;
+         break;
+      }
+   }
+   int totalCoins = 0;
```

Comment:
There is no special case required while writing a test. You should write a general test which compares and prints whether the test passed or failed based on whether _ALL_ cases have passed/failed.